lang/src/OTS/ITF/org/partiql/ots/Plugin.kt

```diff
@@ -1,11 +1,17 @@
 package OTS.ITF.org.partiql.ots
 
 import OTS.ITF.org.partiql.ots.operator.ScalarOp
+import OTS.ITF.org.partiql.ots.type.ScalarType
 
 /**
  * Used to define a plugin
  */
 interface Plugin {
+    /**
+     * All the scalar types
+     */
+    val scalarTypes: List<ScalarType>
+
     val posOp: ScalarOp
     val negOp: ScalarOp
     val binaryPlusOp: ScalarOp
```

Comment:
We need to review the "Plugin" interface based on what it brings to the engine.

- Plugin name spaces
- Scalar Types
- Scalar Functions
- Aggregate Functions
- Plugin specific configuration
- Operator implementations

Are the plugins like Trino's with data sources, udf's, and schema providers? Or are these plugins just for types and functions?

Reply:
For now we only consider there is only one plugin for approaching pluggability of OTS.